# 00 - Dataset introduction

Section **3.1 Introduction** of the report.

## Brief

Describe the dataset: source, what each row/column represents, define the target variable and the type of problem (classification or regression).

- Origin and source of the dataset.
- Description of rows (observational unit) and columns (features).
- Target variable.
- Problem type: classification / regression.

## Source

Dataset: [Wine Reviews](https://www.kaggle.com/datasets/zynicide/wine-reviews/data) — scraped from WineEnthusiast magazine by Zack Thoutt (Kaggle).

Three files are available in `data/raw/`:

| File | Format | Rows | Columns |
|---|---|---|---|
| `winemag-data-130k-v2.csv` | CSV | 129,971 | 13 |
| `winemag-data_first150k.csv` | CSV | 150,930 | 10 |
| `winemag-data-130k-v2.json` | JSON | 129,971 | 13 |

The **v2 CSV** (and its JSON mirror) is the primary file for this project — it is the most complete version, containing taster metadata not present in the 150k file.

In [1]:
import pandas as pd
from pathlib import Path

RAW = Path("../data/raw")

## winemag-data-130k-v2.csv (primary)

129,971 wine reviews. Each row is one review written by a WineEnthusiast taster.

In [2]:
df_v2 = pd.read_csv(RAW / "winemag-data-130k-v2.csv", index_col=0)
print(f"Shape: {df_v2.shape}")
print(f"\nDtypes:\n{df_v2.dtypes}")
print(f"\nMissing values:\n{df_v2.isnull().sum()}")
df_v2.describe(include="all")

Shape: (129971, 13)

Dtypes:
country                   object
description               object
designation               object
points                     int64
price                    float64
province                  object
region_1                  object
region_2                  object
taster_name               object
taster_twitter_handle     object
title                     object
variety                   object
winery                    object
dtype: object

Missing values:
country                     63
description                  0
designation              37465
points                       0
price                     8996
province                    63
region_1                 21247
region_2                 79460
taster_name              26244
taster_twitter_handle    31213
title                        0
variety                      1
winery                       0
dtype: int64


,country,description,designation,points,price,province,region_1,region_2,taster_name,taster_twitter_handle,title,variety,winery
count,129908,129971,92506,129971.000000,120975.000000,129908,108724,50511,103727,98758,129971,129970,129971
unique,43,119955,37979,NaN,NaN,425,1229,17,19,15,118840,707,16757
top,US,"Seductively tart in lemon pith, cranberry and ...",Reserve,NaN,NaN,California,Napa Valley,Central Coast,Roger Voss,@vossroger,Gloria Ferrer NV Sonoma Brut Sparkling (Sonoma...,Pinot Noir,Wines & Winemakers
freq,54504,3,2009,NaN,NaN,36247,4480,11065,25514,25514,11,13272,222
mean,NaN,NaN,NaN,88.447138,35.363389,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,3.039730,41.022218,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,80.000000,4.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,86.000000,17.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,88.000000,25.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,91.000000,42.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


### Column descriptions

| Column | Type | Description | Missing |
|---|---|---|---|
| `country` | categorical | Country where the wine was produced | 63 |
| `description` | text | Taster's written review | 0 |
| `designation` | categorical | Vineyard sub-label within the winery | 37,465 |
| `points` | int | Score assigned by the taster (80–100) | 0 |
| `price` | float | Bottle price in USD | 8,996 |
| `province` | categorical | Province or state of origin | 63 |
| `region_1` | categorical | Specific wine-growing region | 21,247 |
| `region_2` | categorical | More specific sub-region | 79,460 |
| `taster_name` | categorical | Name of the WineEnthusiast reviewer | 26,244 |
| `taster_twitter_handle` | categorical | Twitter handle of the reviewer | 31,213 |
| `title` | text | Full wine label (includes vintage year) | 0 |
| `variety` | categorical | Grape variety (e.g. Pinot Noir, Chardonnay) | 1 |
| `winery` | categorical | Name of the producing winery | 0 |

**Key stats:**
- `points`: mean 88.4, std 3.0, range 80–100
- `price`: mean $35.4, std $41.0, range $4–$3,300 (heavy right skew)
- `country`: 43 unique values — US dominates with 54,504 reviews
- `variety`: 707 unique grape varieties

## winemag-data_first150k.csv (older version)

150,930 wine reviews. An earlier scrape — larger but missing the taster columns present in v2.

In [3]:
df_150k = pd.read_csv(RAW / "winemag-data_first150k.csv", index_col=0)
print(f"Shape: {df_150k.shape}")
print(f"\nDtypes:\n{df_150k.dtypes}")
print(f"\nMissing values:\n{df_150k.isnull().sum()}")
df_150k.describe(include="all")

Shape: (150930, 10)

Dtypes:
country         object
description     object
designation     object
points           int64
price          float64
province        object
region_1        object
region_2        object
variety         object
winery          object
dtype: object

Missing values:
country            5
description        0
designation    45735
points             0
price          13695
province           5
region_1       25060
region_2       89977
variety            0
winery             0
dtype: int64


,country,description,designation,points,price,province,region_1,region_2,variety,winery
count,150925,150930,105195,150930.000000,137235.000000,150925,125870,60953,150930,150930
unique,48,97821,30621,NaN,NaN,455,1236,18,632,14810
top,US,A little bit funky and unsettled when you pop ...,Reserve,NaN,NaN,California,Napa Valley,Central Coast,Chardonnay,Williams Selyem
freq,62397,6,2752,NaN,NaN,44508,6209,13057,14482,374
mean,NaN,NaN,NaN,87.888418,33.131482,NaN,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,3.222392,36.322536,NaN,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,80.000000,4.000000,NaN,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,86.000000,16.000000,NaN,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,88.000000,24.000000,NaN,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,90.000000,40.000000,NaN,NaN,NaN,NaN,NaN


### Differences from v2

| Aspect | v2 (130k) | first150k |
|---|---|---|
| Rows | 129,971 | 150,930 |
| Columns | 13 | 10 |
| `taster_name` | yes | **no** |
| `taster_twitter_handle` | yes | **no** |
| `title` | yes | **no** |
| `points` mean | 88.4 | 87.9 |
| `price` mean | $35.4 | $33.1 |
| `variety` unique | 707 | 632 |

This file will **not** be used for modelling — it is kept for reference only.

## winemag-data-130k-v2.json

Same 129,971 records and 13 fields as the v2 CSV — just a different format. Kept as an alternative ingestion path.

In [4]:
import json

with open(RAW / "winemag-data-130k-v2.json") as f:
    data_json = json.load(f)

print(f"Type: {type(data_json)}")
print(f"Records: {len(data_json)}")
print(f"Fields: {list(data_json[0].keys())}")

Type: <class 'list'>
Records: 129971
Fields: ['points', 'title', 'description', 'taster_name', 'taster_twitter_handle', 'price', 'designation', 'variety', 'region_1', 'region_2', 'province', 'country', 'winery']
